libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -208,11 +210,13 @@ void GCS_MAVLINK::send_battery_status(const AP_BattMonitor &battery, const uint8
 }
 
 // returns true if all battery instances were reported
-bool GCS_MAVLINK::send_battery_status(const AP_BattMonitor &battery) const
+bool GCS_MAVLINK::send_battery_status() const
 {
+    const AP_BattMonitor &battery = AP::battery();
+
     for(uint8_t i = 0; i < battery.num_instances(); i++) {
         CHECK_PAYLOAD_SIZE(BATTERY_STATUS);
-        send_battery_status(battery, i);
```

Comment:
Why not pass the reference here since you have it, instead of making the final send method look up the reference again?